src/App.jsx

```diff
@@ -100,6 +137,14 @@ function App() {
       setMessage(`Error removing todo: ${error.message}`);
     }
   };
+  // Toggle function to switch between ascending and descending
+  const toggleSortOrder = () => {
+    setIsAscending(!isAscending); // Toggle sort order
+  };
+
+  React.useEffect(() => {
+    fetchData();
+  }, [isAscending]); // Refetch data when sort order changes
```

Comment:
The fetchData function is being called in two useEffect hooks, which might not be necessary. You can refactor this to avoid re-fetching the data unnecessarily when toggling the sort order.

You can manage the sorting on the client-side without refetching the data when toggling the sort order. Instead of fetching data again when isAscending changes, you can sort the existing todoList state directly.

Reply:
Completed